tools/file_packager.py

```diff
@@ -965,59 +965,68 @@ def generate_js(data_target, data_files, metadata):
           });
           return;
         }'''.strip()
+
     ret += '''
       function fetchRemotePackage(packageName, packageSize, callback, errback) {
         %(node_support_code)s
-        var xhr = new XMLHttpRequest();
-        xhr.open('GET', packageName, true);
-        xhr.responseType = 'arraybuffer';
-        xhr.onprogress = function(event) {
-          var url = packageName;
-          var size = packageSize;
-          if (event.total) size = event.total;
-          if (event.loaded) {
-            if (!xhr.addedTotal) {
-              xhr.addedTotal = true;
-              if (!Module.dataFileDownloads) Module.dataFileDownloads = {};
-              Module.dataFileDownloads[url] = {
-                loaded: event.loaded,
-                total: size
-              };
-            } else {
-              Module.dataFileDownloads[url].loaded = event.loaded;
+        Module.dataFileDownloads = Module.dataFileDownloads || {};
+        const url = packageName;
+        fetch(url)
+        .catch(error => { throw new Error(error + ' : ' + url) }) // Do this first so we don't catch errors from then()
```

Comment:
Instead of `.catch` here is it the same to write the error handler as the second argument to `then()`?     Is one way preferable over the other?

Reply:
In this case, we need to `catch()` before `then()` so we only handle fetch errors here. If we putt this at the bottom, or in the second param of  `this(),` then it will also act on the network errors generated by `then(),` and I'd need to add logic to skip certain errors.